### DISCUSSION TITLE
Disable serialization for specific types

### DISCUSSION THREAD

**Discussion by JamesTheButler** [aaubry/YamlDotNet#861] [Q&A] [answered]
Hi everyone!

I am serializing a class that contains a public `Action` field. Consider the following example:
```
record class Example
{
  public int Int { get; private set; }

  public Action? SomeAction;
};

public void Test()
{
  var serializer = new SerializerBuilder().Build();
  var output = serializer.Serialize(new DemoClass());
  // output:
  //  SomeAction: 
  //  Int: 0
}
```
`SomeAction` gets serialized to an empty string in yaml, which is a bit annoying but still okay. It get's fun if we actually subscribe to `SomeAction` when the serializer generates ~5.000 lines of yaml to describe the attached action.

Is it possible to **not** serialize members of a given type? I know about `[YamlIgnore]` but I would rather set up my serializer to ignore all members of the type `Action` (and possibly other types as well) in any of my serializable classes.

**Comment by EdwardCooke** [comment, marked as answer]
I just made this an issue. 

And indeed, actions and funcs should probably be ignored. I’ll put up a new version shortly. I do like the idea of also adding an ignore type feature. That could be neat and I’ll build it with that in mind

**Comment by EdwardCooke** [comment]
Sorry it's taking so long, personal and work stuff has been taking priority. I'll try and get to this in the next week or 2. I have added an issue for this so I'm going to close the discussion.

#862
